tests/test_entrypoint.py

```diff
@@ -0,0 +1,90 @@
+from pathlib import Path
+from google.cloud import bigquery
+import os
+import pytest
+import subprocess
+
+ENTRYPOINT_SCRIPT = Path(__file__).parent.parent / "script" / "entrypoint"
+
+
+class TestEntrypoint:
+    @pytest.mark.integration
+    def test_run_query(self, tmp_path, project_id):
+        query_file_path = tmp_path / "sql" / "test" / "query_v1"
+        os.makedirs(query_file_path)
+
+        query_file = query_file_path / "query.sql"
+        query_file.write_text("SELECT 1 AS a, 'abc' AS b;")
+
+        try:
+            result = subprocess.check_output(
+                [
+                    ENTRYPOINT_SCRIPT,
+                    "query",
+                    "--project_id=" + project_id,
+                    str(query_file),
+                ],
+                stderr=subprocess.STDOUT,
+            )
+            assert b"Current status: DONE" in result
+            assert b"No metadata.yaml found for {}" in result
+        except subprocess.CalledProcessError as e:
+            # running bq in CircleCI will fail since it's not installed
+            # but the error output can be checked for whether bq was called
+            assert b"No such file or directory: 'bq'" in e.output
+            assert b"No metadata.yaml found for {}" in e.output
+            assert (
+                b'subprocess.check_call(["bq"] + query_arguments, stdin=query_stream)'
+                in e.output
+            )
```

Comment:
I tried installing the cloud SDK and bq in CircleCI but get the following error:

```
ERROR: gcloud failed to load: While reading from '/root/.config/gcloud/configurations/config_default' [line  5]: option 'project' in section 'core' already exists
    gcloud_main = _import_gcloud_main()
    import googlecloudsdk.gcloud_main
    from googlecloudsdk.api_lib.iamcredentials import util as iamcred_util
    from googlecloudsdk.api_lib.util import apis_internal
    from googlecloudsdk.core import transport
    from googlecloudsdk.core import log
    _log_manager = _LogManager()
    self.Reset(sys.stdout, sys.stderr)
    std_console_formatter = _ConsoleFormatter(stderr)
    use_color = not properties.VALUES.core.disable_color.GetBool(validate=False)
    named_configs.ActivePropertiesFile.Load(),
    ActivePropertiesFile._PROPERTIES = properties_file.PropertiesFile(
    self.__Load(properties_path)
    parsed_config.read(properties_path)
    self._read(fp, filename)
    raise DuplicateOptionError(sectname, optname,

This usually indicates corruption in your gcloud installation or problems with your Python interpreter.
```

So what I ended up doing here is checking if `bq` runs correctly, or if it is not installed, if the error output is as expected, which should also include the `subprocess.check_call` to `bq`.

I checked some of the recent issues that resulted in Airflow failures, these tests would have caught those.